common/src/sql/dbinit.sql

```diff
@@ -107,14 +107,10 @@ CREATE TABLE omicron.public.sled (
 );
 
 /* Add an index which lets us look up sleds on a rack */
-CREATE INDEX ON omicron.public.sled (
-    rack_id
-) WHERE time_deleted IS NULL;
-
-CREATE INDEX ON omicron.public.sled (
+CREATE UNIQUE INDEX ON omicron.public.sled (
+    rack_id,
     id
-) WHERE
-    time_deleted IS NULL;
+) WHERE time_deleted IS NULL;
```

Comment:
This is one of the cases mentioned in the PR description.  If we want to paginate the sleds in a rack, we need the sled id in this index, and then it can also be unique.